master/static/views_and_triggers/up/workspaces.sql

```diff
@@ -0,0 +1,17 @@
+CREATE OR REPLACE FUNCTION prevent_auto_created_namespace_change() RETURNS TRIGGER AS $$
+        BEGIN
+                RAISE integrity_constraint_violation 
+                        USING MESSAGE = 'Cannot change auto_created_namespace_name once set';
+        END;
+$$ LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS on_update_auto_created_namespace_name ON workspaces CASCADE;
+
+-- We want to enforce immutability of a workspace's auto_created_namespace_name.
+CREATE TRIGGER on_update_auto_created_namespace_name
+   BEFORE UPDATE ON workspaces
+   FOR EACH ROW
+   WHEN (OLD.auto_created_namespace_name != NEW.auto_created_namespace_name
+        OR (NEW.auto_created_namespace_name IS NULL) 
+        AND OLD.auto_created_namespace_name IS NOT NULL)
```

Comment:
Maybe it's just me, but it took me a bit to unpack this and wrap my head around it. Might make it easier to reason about if we start with the headline, so to speak, of "we only care about the case when it starts off `NOT NULL`.
```suggestion
   WHEN OLD.auto_created_namespace_name IS NOT NULL AND (
        OLD.auto_created_namespace_name != NEW.auto_created_namespace_name
        OR NEW.auto_created_namespace_name IS NULL
   )
```

Reply:
Good point!

Reply:
I'll use 
```  
 WHEN (OLD.auto_created_namespace_name IS NOT NULL AND OLD.auto_created_namespace_name
     IS DISTINCT FROM  NEW.auto_created_namespace_name)
```
since it's a little bit shorter (and hopefully clearer? Or do you think the longer query is clearer)